src/model/sample.rs

```diff
@@ -340,7 +349,8 @@ impl Sample {
                     let pos = record.pos() as u32;
                     let end_pos = cigar.end_pos() as u32;
 
-                    let overlap = {
+                    let overlap = { // BUG: SoftClips can be preceded by HardClips at read starts, followed by HardClips at read ends!
+                        // TODO: suggestion for removing the above BUG: consider this in the CigarStringView::end_pos calculation by adding in the include_softclips argument there, as well, and create an according function CigarStringView::start_pos
```

Comment:
Actually this is intended. I am only interested in the part of the read where the sequence is known and real. Hard clipped bases are not in the bam. So, I think the comments can be removed.